### PR TITLE
Show confirmation before user closes post without saving

### PR DESCRIPTION
**Pull request by Devorein**
image

### CODE REVIEW

File: components/forum/components/PostDialog/PostDialog.tsx [hunk 1]
```diff
@@ -72,9 +82,38 @@ export default function PostDialog({ post, spaceId, onClose, open }: Props) {
...
+        } else {
+          close();
+        }
+      }}
     >
-      <PostPage post={post ?? null} spaceId={spaceId} onSave={close} />
+      <PostPage
+        form={form}
+        setForm={(_form) => {
+          const updatedFormInputs = typeof _form === 'function' ? _form(form) : _form;
```
Comment (msadura) [inline components/forum/components/PostDialog/PostDialog.tsx:96]: This brings some questions :D Why `_form` can be a function? This looks like some hacks.
Looks like I missed the part of storing `contentText` - why is this needed?

Comment (Devorein) [reply components/forum/components/PostDialog/PostDialog.tsx]: Changed it to be less complex. Since it is a state setter it could be a function `(prevState) => newState` or just `newState`

File: components/forum/components/PostDialog/PostDialog.tsx [hunk 2]
```diff
@@ -72,9 +76,35 @@ export default function PostDialog({ post, spaceId, onClose, open }: Props) {
...
+      <ConfirmDeleteModal
+        onClose={() => {
+          setShowConfirmDialog(false);
+        }}
+        title='Unsaved changes'
+        open={showConfirmDialog}
+        buttonText='Discard'
+        secondaryButtonText='Go back'
+        question='Are you sure you want to close this post? You have unsaved changes'
+        onConfirm={close}
```
Comment (motechFR) [inline components/forum/components/PostDialog/PostDialog.tsx:106]: An import UX consideration; When I click discard, and leave the post. I'd expect the content to be deleted.

However, when I come back to the post, the content is still there

**Review by motechFR — APPROVED** [review]
Does what it says, no further comment

File: components/forum/components/PostDialog/PostDialog.tsx (hunk 2 above)
Comment (Devorein) [reply components/forum/components/PostDialog/PostDialog.tsx:106]: Thank you for catching that. Will fix 💯

**Review by valentinludu — COMMENTED** [review]
If -> I write a title, cat, desc -> click close -> click discard -> Click Create new post

I see the Create new post modal with the title and description from before and no category selected. Is that intentional or a bug? I would have expect to see all empty.

**Review by valentinludu — COMMENTED** [review]
Probably a bug:

IF -> click create new post -> fill the fields in this exact order: category, title, content.

The POST button is disabled and the tooltip says: Title is required. But title is filled. :(

**Review by valentinludu — CHANGES_REQUESTED** [review]
Some small bugs

**Comment by Devorein** [comment]
That was quite difficult. Thank you for spotting the bug. I had to spend quite a lot of time figuring out the issues and fixing them.
